common/model/transaction.c

```diff
@@ -414,6 +414,22 @@ void transaction_set_hash(iota_transaction_t transaction,
   memcpy(transaction->hash, hash, sizeof(transaction->hash));
 }
 
+// Reset all transaction fields
+void transaction_reset(iota_transaction_t transaction) {
+  memset(transaction, 0, sizeof(struct _iota_transaction));
+  memset(transaction->signature_or_message, FLEX_TRIT_NULL_VALUE,
+         sizeof(transaction->signature_or_message));
+  memset(transaction->obsolete_tag, FLEX_TRIT_NULL_VALUE,
+         sizeof(transaction->obsolete_tag));
+  memset(transaction->bundle, FLEX_TRIT_NULL_VALUE,
+         sizeof(transaction->bundle));
+  memset(transaction->trunk, FLEX_TRIT_NULL_VALUE, sizeof(transaction->trunk));
+  memset(transaction->branch, FLEX_TRIT_NULL_VALUE,
+         sizeof(transaction->branch));
+  memset(transaction->tag, FLEX_TRIT_NULL_VALUE, sizeof(transaction->tag));
+  memset(transaction->nonce, FLEX_TRIT_NULL_VALUE, sizeof(transaction->nonce));
+}
```

Comment:
What about the address and hash ?

Reply:
Following a discussion with Andreas, it was decided to leave them untouched.